scripts/json_overview_image_info.py

```diff
@@ -18,19 +18,33 @@
 
 output = {}
 
+
+def get_initial_output(image_info):
+    # preserve existing profiles.json
+    if output_path.is_file():
+        profiles = json.loads(output_path.read_text())
+        if profiles["version_code"] == image_info["version_code"]:
+            return profiles
+    return image_info
+
+
 for json_file in work_dir.glob("*.json"):
     image_info = json.loads(json_file.read_text())
+
     if not output:
-        output.update(image_info)
+        output = get_initial_output(image_info)
```

Comment:
This will open, read and parse the profiles.json file every time and also doesn't update it between reads if I'm not mistaken. Meaning the `output` variable is "reset" on every iteration to the "initial output" rather than being extended.

Did you try this patch with 3 or more profiles?